lib/libzfs/libzfs_dataset.c

```diff
@@ -196,6 +196,16 @@ zfs_validate_name(libzfs_handle_t *hdl, const char *path, int type,
 				zfs_error_aux(hdl, dgettext(TEXT_DOMAIN,
 				    "reserved disk name"));
 				break;
+			
+			case NAME_ERR_SELF_REFERENCE:
+				zfs_error_aux(hdl, dgettext(TEXT_DOMAIN,
+				    "self reference is found"));
+				break;
+			
+			case NAME_ERR_PARENT_REFERENCE:
+				zfs_error_aux(hdl, dgettext(TEXT_DOMAIN,
+				    "parent reference is found"));
```

Comment:
same here:  `"dot-dot ('..') component in name"`

Reply:
parent reference, ('..') is found in name